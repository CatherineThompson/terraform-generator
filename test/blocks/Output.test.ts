import { Output } from '../..';
import { arg4 } from '.';

test('Output', () => {
  const output = new Output('name', arg4);
  expect(output.toTerraform('0.11')).toMatchSnapshot();
  expect(output.toTerraform('0.12')).toMatchSnapshot();
  expect(() => output.asArgument()).toThrow();
  expect(() => output.getAttribute('attr')).toThrow();
});
