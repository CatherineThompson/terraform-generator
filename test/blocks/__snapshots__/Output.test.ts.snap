// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output 1`] = `
"output \\"name\\"{
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg14 = {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = \\"\${type.name.attr}\\"
arg7 = [
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
\\"\${type.name.attr}\\",
]
arg8 = \\"\${type.name}\\"
arg9 = [
\\"\${type.name}\\",
\\"\${type.name}\\",
\\"\${type.name}\\",
]
arg10 = [
\\"e\\",
4,
true,
\\"\${type.name.attr}\\",
\\"\${type.name}\\",
]
arg11 = \\"\${max(5, 12, 9)}\\"
}
}
}
}

"
`;

exports[`Output 2`] = `
"output \\"name\\"{
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg14 = {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
arg13 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
arg12 {
arg0 = \\"a\\"
arg1 = [
\\"b\\",
\\"c\\",
\\"d\\",
]
arg2 = 0
arg3 = [
1,
2,
3,
]
arg4 = true
arg5 = [
false,
true,
false,
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr,
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name,
]
arg10 = [
\\"e\\",
4,
true,
type.name.attr,
type.name,
]
arg11 = max(5, 12, 9)
}
}
}
}

"
`;
